### PR TITLE
Contexthandler: Remove code that is no longer used

### PR DESCRIPTION
**Pull request by kalleep**
**What is this feature?**
Now that Auth broker always is enabled there are a lot of code we can remove.

This pr removes code from context handler that we are no longer using and adding some tests.

Most changes in other places are test setups

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/authn/clients/password.go [hunk 1]
```diff
@@ -12,9 +12,9 @@ import (
 )
 
 var (
-	errEmptyPassword       = errutil.NewBase(errutil.StatusBadRequest, "password-auth.empty", errutil.WithPublicMessage("Invalid username or password"))
-	errPasswordAuthFailed  = errutil.NewBase(errutil.StatusBadRequest, "password-auth.failed", errutil.WithPublicMessage("Invalid username or password"))
-	errInvalidPassword     = errutil.NewBase(errutil.StatusBadRequest, "password-auth.invalid", errutil.WithPublicMessage("Invalid password or username"))
+	errEmptyPassword       = errutil.NewBase(errutil.StatusUnauthorized, "password-auth.empty", errutil.WithPublicMessage("Invalid username or password"))
```
Comment (Jguer) [inline pkg/services/authn/clients/password.go:15]: were they not synced with context handler?

Comment (Jguer) [reply pkg/services/authn/clients/password.go:15]: or is it to fix enumeration?

Comment (kalleep) [reply pkg/services/authn/clients/password.go:15]: No, wrong status code

Comment (Jguer) [reply pkg/services/authn/clients/password.go:15]: I think there's an issue you can close if that's the case

**Review by Jguer — APPROVED** [review]

**Pull request merged**